### PR TITLE
chore(RHTAPWATCH-160): Run SAST checks for each opened PR

### PR DESCRIPTION
**Pull request by Kousalya1998**
# Description

- Run SAST checks for each opened PR

## Issue ticket number and link
[RHTAPWATCH-160](https://issues.redhat.com/browse/RHTAPWATCH-160)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added meaningful description with JIRA/GitHub issue key (if applicable)
- [x] I have updated labels (if needed)

### CODE REVIEW

**Review by hmariset — COMMENTED** [review]
new line missing at end of file

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ repos:
   - hooks:
       - id: yamllint
     repo: https://github.com/adrienverge/yamllint
-    rev: v1.29.0
+    rev: v1.32.0
```
Comment (Omeramsc) [inline .pre-commit-config.yaml:6]: Is this required for the SAST checks? if not, omit it from this commit.

**Comment by ifireball** [comment]
please rebase and resolve conflicts

**Comment by Omeramsc** [comment]
@Kousalya1998 it seems like the pre-commit checks are not running\stuck. 
also you should resolve the conflict with the `pre-commit-config` file, it may help.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -19,3 +19,7 @@ repos:
       - id: go-mod-tidy
     repo: https://github.com/dnephin/pre-commit-golang
     rev: v0.5.1
+  - hooks:
+      - id: snyk-iac
+        args: ["test", "-d", "$(pwd)"]
+    repo: https://github.com/fabasoad/pre-commit-snyk
```
Comment (Omeramsc) [inline .pre-commit-config.yaml:25]: Missing `rev` key.

**Review by ifireball — CHANGES_REQUESTED** [review]

**Comment by Kousalya1998** [comment]
@Omeramsc I resolved the conflicts but the CI keeps failing

**Comment by yftacherzog** [comment]
> @Omeramsc I resolved the conflicts but the CI keeps failing

one failure is gitlint. You need to fix the commit message (missing colon after the closing brackets)

**Comment by Kousalya1998** [comment]
> > @Omeramsc I resolved the conflicts but the CI keeps failing
> 
> one failure is gitlint. You need to fix the commit message (missing colon after the closing brackets)

Updated. Thanks.

**Comment by yftacherzog** [comment]
> > @Omeramsc I resolved the conflicts but the CI keeps failing
> 
> one failure is gitlint. You need to fix the commit message (missing colon after the closing brackets)

The second one is about the revision. It tells which release of the pluging repo should be taken. You need to open the URL mentioned under `repo` and on the right you'd see `Releases`. You need to take the latest one from there. The one you have at the moment does not exist.

**Comment by Kousalya1998** [comment]
> > > @Omeramsc I resolved the conflicts but the CI keeps failing
> > 
> > 
> > one failure is gitlint. You need to fix the commit message (missing colon after the closing brackets)
> 
> The second one is about the revision. It tells which release of the plugin repo should be taken. You need to open the URL mentioned under `repo` and on the right you'd see `Releases`. You need to take the latest one from there. The one you have at the moment does not exist.

Okay added v0.5.0

**Comment by yftacherzog** [comment]
@Kousalya1998 still need to fix multiple CI failures.

**Review by Omeramsc — CHANGES_REQUESTED** [review]
```
go-sec-mod...............................................................Failed
- hook id: go-sec-mod
- duration: 0.01s
- exit code: 1

error: command not found: gosec
```

seems like this check still fails

**Comment by Kousalya1998** [comment]
> ```
> go-sec-mod...............................................................Failed
> - hook id: go-sec-mod
> - duration: 0.01s
> - exit code: 1
> 
> error: command not found: gosec
> ```
> 
> seems like this check still fails

Updated the code. Ignore this version.

**Comment by Kousalya1998** [comment]
> ```
> go-sec-mod...............................................................Failed
> - hook id: go-sec-mod
> - duration: 0.01s
> - exit code: 1
> 
> error: command not found: gosec
> ```
> 
> seems like this check still fails

snyk check passes now.

**Comment by Kousalya1998** [comment]
@Omeramsc @hmariset @amisstea Need your reviews.

**Review by Omeramsc — APPROVED** [review]
LGTM

**Review by hmariset — APPROVED** [review]
LGTM

**Comment by ifireball** [comment]
@Kousalya1998 assuming the SASK task image version is being updated by renovatebot like the other images in the pipeline to we really still need this PR?

**Comment by Kousalya1998** [comment]
> assuming the SASK task image version is being updated by renovatebot like the other images in the pipeline do we really still need this PR?

As far as I can tell, the SAST check passed now and that was the requirement for this task. I am not sure about renovatebot..

**Comment by Omeramsc** [comment]
Where does this change stands? Are we planning to use it?
(If so, please rebase and solve conflicts)

File: .tekton/segment-bridge-pull-request.yaml [hunk 3]
```diff
@@ -309,7 +309,7 @@ spec:
         - name: name
           value: sast-snyk-check
         - name: bundle
-          value: quay.io/redhat-appstudio-tekton-catalog/task-sast-snyk-check:0.1@sha256:f57c69cfedf37a46dd61285b5b9b4805ff196facaad2d7cdeded496c77d31bec
+          value:  quay.io/redhat-appstudio-tekton-catalog/task-sast-snyk-check:0.1@sha256:edd4ba638b71de52c2662abd3e93fd876e6e75cd07b162d13fae014d3a1a1fac
```
Comment (amisstea) [inline .tekton/segment-bridge-pull-request.yaml:312]: Please remove leading spaces.

**Review by amisstea — APPROVED** [review]
I don't see why we shouldn't merge this. The `.snyk` file seems necessary and renovatebot hasn't updated these container image refs for whatever reason.

File: .tekton/segment-bridge-pull-request.yaml (hunk 3 above)
Comment (Kousalya1998) [reply .tekton/segment-bridge-pull-request.yaml]: Removed.

**Comment by ifireball** [comment]
I can see the SNYK check is running now:

Screenshot from 2023-12-07 10-04-12

But its failing with these errors in the logs:
```
[sast-snyk-check] sast-snyk-check test failed because of the following issues:
[sast-snyk-check] Snyk Code is not supported for org: enable in Settings > Snyk Code
[sast-snyk-check] {"result":"ERROR","timestamp":"1701929575","note":"Task sast-snyk-check failed: For details, check Tekton task log.","namespace":"default","successes":0,"failures":0,"warnings":0}
```

What will it take to make it pass?

**Comment by Kousalya1998** [comment]
> I can see the SNYK check is running now:
> 
> Screenshot from 2023-12-07 10-04-12
> 
> But its failing with these errors in the logs:
> 
> ```
> [sast-snyk-check] sast-snyk-check test failed because of the following issues:
> [sast-snyk-check] Snyk Code is not supported for org: enable in Settings > Snyk Code
> [sast-snyk-check] {"result":"ERROR","timestamp":"1701929575","note":"Task sast-snyk-check failed: For details, check Tekton task log.","namespace":"default","successes":0,"failures":0,"warnings":0}
> ```
> 
> What will it take to make it pass?

@ifireball I saw that too. So, to make it `SUCCESS` we need that Snyk subscription account. I am trying to get that access. It throws an error because I used a free trial version.

**Comment by Kousalya1998** [comment]
@ifireball @amisstea I FINALLY  got the Red Hat SSO access to the SNYK tool!! updated the secrets. However, it still gives the ERROR message. Something else is wrong now. Checking on that with the `#rhtap developers` channel.
image

**Comment by Kousalya1998** [comment]
Similar issues have arisen with the 'service-provider integrator' team, who have claimed they merged the PR without any issues even after the FAILURE message. Testing it.
https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1702020501097879

**Comment by Kousalya1998** [comment]
@ifireball I had a meeting with Scott Hebert today and we came across an image that we are using in the segment-bridge's Dockerfile:
```
# Second stage: Create the final container image
FROM registry.redhat.io/openshift4/ose-tools-rhel8:v4.13.0-202311211131.p0.gc7c6eb2.assembly.stream
```

Also, I see that tools for [Red Hat appstudio's Dockerfile ](https://github.com/redhat-appstudio/tools/blob/main/Dockerfile) use a different image which looks like:
```
RUN \
    curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.12.36/openshift-client-linux.tar.gz \
    | tar -C /opt/app-root/bin/ -xvzf - oc \
    && /usr/libexec/s2i/assemble
```
I guess the image that we are using causes HIGH vulnerability and that is making the SAST check Fail. However, with the tools repo if you see the SAST check passes with a different image:
image

**Comment by ifireball** [comment]
> @ifireball I had a meeting with Scott Hebert today and we came across an image that we are using in the segment-bridge's Dockerfile:
> 
> ```
> # Second stage: Create the final container image
> FROM registry.redhat.io/openshift4/ose-tools-rhel8:v4.13.0-202311211131.p0.gc7c6eb2.assembly.stream
> ```
> 
> Also, I see that tools for [Red Hat appstudio's Dockerfile ](https://github.com/redhat-appstudio/tools/blob/main/Dockerfile) using a different image which looks like:
> 
> ```
> RUN \
>     curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.12.36/openshift-client-linux.tar.gz \
>     | tar -C /opt/app-root/bin/ -xvzf - oc \
>     && /usr/libexec/s2i/assemble
> ```
> 
> I guess the image that we are using causes HIGH vulnerability and that is making the SAST check Fail. However, with the tools repo if you see the SAST check passes with a different image: image

I think the issues Scott talked about have to do with the Clair scan and not the SAST scan. From looking at the logs it seems the SAST scan is failing on issues in the `kwok/host-operator` directory which is a Git submodule linking to the [host-operator](https://github.com/codeready-toolchain/host-operator) repo that we don't own. We can't do much about SAST issue in it.

If the SAST scan is running and failing on actual security issues v.s. system/invocation issues we can regard this task as successful.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: .snyk [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.
+version: v1.25.0
+# ignores vulnerabilities until expiry date; change duration by modifying expiry date
+exclude:
+  global:
```
[the remark this answers is not in the document]
Comment (Kousalya1998) [reply .snyk:5]: Got it. reverted the changed and added it under exclude

File: .tekton/segment-bridge-pull-request.yaml [hunk 5]
```diff
@@ -224,7 +228,7 @@ spec:
         - name: name
           value: buildah
         - name: bundle
-          value: quay.io/redhat-appstudio-tekton-catalog/task-buildah:0.1@sha256:d95e898c518093c24bc7dacd25aa98a39996262a6b34650e403b60b1c8c10fda
+          value: quay.io/redhat-appstudio-tekton-catalog/task-buildah:0.1@sha256:9e7be6d3f2cde33fabfda9d454757530ce7f73e2414c00da1a3ebabf95fc52c5
```
[the remark this answers is not in the document]
Comment (Kousalya1998) [reply .tekton/segment-bridge-pull-request.yaml]: reverted the changes I made to the images. Still gives some errors

**Comment by klakshma21** [comment]
Due to losing access to my current GitHub account, I am closing this PR and opening a new one
https://github.com/redhat-appstudio/segment-bridge/pull/147

**Comment by Omeramsc** [comment]
Closing as a new PR opened instead.

**Pull request closed without merging**